Update Music Assistant documentation to the Gold Quality Level (#38323)
home-assistant/home-assistant.github.io: source/_integrations/music_assistant.markdown

  - **Description**: Start playing the queue on the target player. Omit to use the default behavior.
  - **Example**: `true`

#### Example

In this example, the queue of the first player found playing will be transferred to the kitchen when a motion sensor is triggered in that room.

```yaml
automation:
  - id: auto_queue_transfer_kitchen
    alias: Automatically Transfer Queue to Kitchen
    trigger:
      platform: state
      entity_id: binary_sensor.kitchen_motion_sensor_occupancy
      to: 'on'
    action:
      service: music_assistant.transfer_queue
      target:
        entity_id: media_player.ma_kitchen_speaker
```

### Action `music_assistant.search`

Perform a global search on the Music Assistant library and all providers. This allows programmatic access to all of the music provider's catalogs and could be used to build a HA dashboard where any track could be found for playback.

- **Data attribute**: `config_entry_id`
  - **Optional**: No.

>   - **Description**: The Music Assistant instance that the search will be performed upon. Allows for multiple servers to be running. This is obtained from a dropdown in the GUI editor. Users of yaml can use the dev tools action tab and select from the dropdown and then switch to yaml to get the actual value.

Reviewer: ```suggestion
  - **Description**: The Music Assistant instance that the search will be performed upon. Allows for multiple servers to be running. This is obtained from a dropdown in the GUI editor. Users of YAML can use the dev tools action tab and select from the dropdown and then switch to YAML to get the actual value.
```

- **Data attribute**: `entity_id`
  - **Optional**: No.
  - **Description**: The entity_id of the player holding the queue to be retrieved.
  - **Example**: `media_player.kitchen_speaker`

#### Example

This example sets the name of the currently playing track in an [`input_text`](https://www.home-assistant.io/integrations/input_text/) which could then be used on a dashboard.

```yaml
script:
  get_now_playing:
    mode: queued
    alias: "Get Now Playing Track Name"
    sequence:
      - action: music_assistant.get_queue
        data:
          entity_id: media_player.ma_kitchen_speaker
        response_variable: queue_info
      - service: input_text.set_value
        data:
          entity_id: input_text.now_playing
          value: "{{ queue_info['media_player.ma_kitchen_speaker'].current_item.name }}"
```
  
## Notes

- Any Home Assistant players added to Music Assistant will appear duplicated as the MA version of the player is created. The original HA player can be hidden if desired.

## Known limitations

> The `get_queue` action only returns the current and next item in the queue. This is because a large amount of data is returned and if this was done for potentially thousands of tracks this could have an adverse impact on HA performance or stability.

Reviewer: ```suggestion
The `get_queue` action only returns the current and next item in the queue. This is because a large amount of data is returned, and if this was done for potentially thousands of tracks, this could have an adverse impact on Home Assistant's performance or stability.
```

> The data returned by the `get_queue` action will be partially limited if the item is not in the library (e.g. an item was selected for playback directly from Spotify). Metadata such as favorite status, explicit status, last played, played count and disc art URL are only available for items which are in the MA library.

Reviewer: ```suggestion
The data returned by the `get_queue` action will be partially limited if the item is not in the library (For example, if an item was selected for playback directly from Spotify). Metadata such as favorite status, explicit status, last played, played count, and disc art URL are only available for items that are in the MA library.
```

  - **Description**: When `album` is the `media_type` then this option will restrict the result according to the selection of either album, single, compilation, EP or unknown.
  - **Example**: `album`

#### Example

This example will start playback of ten random tracks.

```yaml
script:
  create_random_queue:
    mode: single
    sequence:
      - service: music_assistant.get_library
        data:
          limit: 10
          media_type: track
          config_entry_id: 01JEXNDHT21V0BHJXM7A5SZANV
          order_by: random
        response_variable: random_tracks
      - repeat:
          count: "{{ random_tracks['items'] | length }}"
          sequence:
            - action: music_assistant.play_media
              data:
                media_id: "{{ random_tracks['items'][repeat.index-1].uri }}"
                media_type: track
                enqueue: add
              target:

>                 entity_id: media_player.ma_kitchen_speaker

Reviewer 2: Why not simply assign the result in a new array of just uri's ?

Author: I could. I have used that somewhere myself.

Author: Code updated

```

### Action `music_assistant.get_queue`

Get the queue details of a Music Assistant player queue. This provides programmatic access to comprehensive information about the current and next media item in the queue. This information could be used to create a bespoke media dashboard.

- **Data attribute**: `entity_id`
  - **Optional**: No.
  - **Description**: The entity_id of the player holding the queue to be retrieved.
  - **Example**: `media_player.kitchen_speaker`

#### Example

This example sets the name of the currently playing track in an [`input_text`](https://www.home-assistant.io/integrations/input_text/) which could then be used on a dashboard.

```yaml
script:
  get_now_playing:
    mode: queued
    alias: "Get Now Playing Track Name"
    sequence:
      - action: music_assistant.get_queue
        data:
          entity_id: media_player.ma_kitchen_speaker
        response_variable: queue_info
      - service: input_text.set_value
        data:
          entity_id: input_text.now_playing
          value: "{{ queue_info['media_player.ma_kitchen_speaker'].current_item.name }}"
```

>   

Reviewer 2: I'm not sure if these examples are needed ?
Maybe better to simply make an example of the command and what it outputs ?

Author: The quality level item says "The documentation provides automation examples the user can use" so I was trying to find examples that met this. It does also mention using blueprints and linking to those but these seemed small enough and I had a problem with one of them trying to get the HA GUI automation editor to accept the template. Lastly, many other HA pages have examples like this on the page.

## Notes

- Any Home Assistant players added to Music Assistant will appear duplicated as the MA version of the player is created. The original HA player can be hidden if desired.

## Known limitations

The `get_queue` action only returns the current and next item in the queue. This is because a large amount of data is returned and if this was done for potentially thousands of tracks this could have an adverse impact on HA performance or stability.

> 

Reviewer 2: I don't see why this is valid info here

Author: Just because people ask can they get the full queue listing and this explains why they can't with the `get_queue` action

Reviewer 2: If they want that, we should introduce a "get_queue_items" - still the text here should be removed.

Author: text removed

The data returned by the `get_queue` action will be partially limited if the item is not in the library (e.g. an item was selected for playback directly from Spotify). Metadata such as favorite status, explicit status, last played, played count and disc art URL are only available for items which are in the MA library.

Radio mode is only available with certain music providers, and an error will be shown if attempting to enable radio mode on an item that isn't linked to one of those providers. Review the Music Assistant documentation to identify which providers support this functionality.

## Troubleshooting

### Can’t find the MA actions

#### Symptom: No Music Assistant actions are shown in the editor

When trying to set up a script or automation via the GUI, no MA actions can be found.

##### Description

This means the add-on may have been installed, but the integration has not.

##### Resolution

Go to the [Configuration section](https://www.home-assistant.io/integrations/music_assistant/#configuration) and install the integration.

> 

Reviewer 2: isn't this a bit obvious when looking at it from the integration docs ?

Author: I know but the quality level requirement said to include troubleshooting and this is all I could think of! I can see someone jumping straight to this when searching for a solution to this problem and we do have people asking this exact question. Happy for a better idea for this though!

Author: Discussed and resolved offline
